src/avro_json_encoder.erl

```diff
@@ -151,18 +176,18 @@ do_encode_type(#avro_record_type{} = T, EnclosingNamespace) ->
   {Name, NextLevelEnclosingNs} = avro:split_type_name(T, Namespace),
   SchemaObjectFields =
     [ optional_field(namespace, ns(Namespace, EnclosingNamespace),
-                     ?NS_GLOBAL, fun encode_string/1)
-    , {type,   encode_string(?AVRO_RECORD)}
-    , {name,   encode_string(Name)}
-    , optional_field(doc,       Doc,  ?NO_DOC, fun encode_string/1)
-    , optional_field(aliases,   Aliases,   [], fun encode_aliases/1)
+                     ?NS_GLOBAL, fun encode_string/1, Opt)
+    , {name, encode_name(Name, canon_ns(Namespace, EnclosingNamespace), Opt)}
+    , {type, encode_string(?AVRO_RECORD)}
+    , optional_field(doc,       Doc,  ?NO_DOC, fun encode_string/1, Opt)
+    , optional_field(aliases,   Aliases,   [], fun encode_aliases/1, Opt)
     , {fields, lists:map(fun(F) ->
-                             encode_field(F, NextLevelEnclosingNs)
+                             encode_field(F, NextLevelEnclosingNs, Opt)
                          end, Fields)}
     | CustomProps
```

Comment:
`CustomProps` should not be added for canonical form.

Reply:
Removed custom props everywhere, now using function head matching instead of conditional code in function bodies.